pkg/authorization/admission/clusterquota/admission.go

```diff
@@ -0,0 +1,116 @@
+package clusterquota
+
+import (
+	"io"
+	"strings"
+	"sync"
+	"time"
+
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+
+	"k8s.io/kubernetes/pkg/admission"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/quota"
+	"k8s.io/kubernetes/pkg/quota/install"
+
+	oclient "github.com/openshift/origin/pkg/client"
+	oadmission "github.com/openshift/origin/pkg/cmd/server/admission"
+)
+
+func init() {
+	admission.RegisterPlugin("ClusterResourceQuota",
+		func(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+			registry := install.NewRegistry(client)
+			return NewResourceQuota(client, registry, 5)
+		})
+}
+
+// quotaAdmission implements an admission controller that can enforce quota constraints
+type quotaAdmission struct {
+	*admission.Handler
+
+	evaluator *quotaEvaluator
+
+	startup            sync.Once
+	clusterQuotaClient oclient.Interface
+	registry           quota.Registry
+	numEvaluators      int
+}
+
+var _ = oadmission.WantsOpenshiftClient(&quotaAdmission{})
+
+type liveLookupEntry struct {
+	expiry time.Time
+	items  []*api.ResourceQuota
+}
+
+// NewResourceQuota configures an admission controller that can enforce quota constraints
+// using the provided registry.  The registry must have the capability to handle group/kinds that
+// are persisted by the server this admission controller is intercepting
+func NewResourceQuota(client clientset.Interface, registry quota.Registry, numEvaluators int) (admission.Interface, error) {
+	return &quotaAdmission{
+		Handler:       admission.NewHandler(admission.Create, admission.Update),
+		registry:      registry,
+		numEvaluators: numEvaluators,
+	}, nil
+}
+
+// Admit makes admission decisions while enforcing quota
+func (q *quotaAdmission) Admit(a admission.Attributes) (err error) {
+	q.startup.Do(func() {
+		evaluator, err := newQuotaEvaluator(q.clusterQuotaClient, q.registry)
```

Comment:
> Why launching it here? Can it be moved in startControllers()?

No, this isn't a controller controller.  Its a controller like backing an admission plugin.